x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import fetch from 'node-fetch';
+import querystring from 'querystring';
+import { Logger } from 'src/core/server';
+import { RequestHandlerContext, KibanaRequest, KibanaResponseFactory } from 'src/core/server';
+import { ConfigType } from '../index';
+
+interface IAppSearchRequestParams<ResponseBody> {
```

Comment:
We should rename AppSearch to EnterpriseSearch, since I plan on refactoring all our current API endpoints to reuse this helper after this PR merges :)

```suggestion
interface IEnterpriseSearchRequestParams<ResponseBody> {
```